management/WFCORE-6830_Management_Resource_Constraints.adoc

```diff
@@ -0,0 +1,221 @@
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+Four new attributes will be added to the HTTP management interface to configure the supported resource constraints that
+can be applied:
+
+ * `backlog`
+ * `connection-high-water`
+ * `connection-low-water`
+ * `no-request-timeout`
+
+ The resulting resource will look like:
+
+[source]
+----
+ /core-service=management/management-interface=http-interface:read-resource(include-defaults=true)
+{
+    "outcome" => "success",
+    "result" => {
+        "allowed-origins" => undefined,
+        "backlog" => 50,
+        "connection-high-water" => 100,
+        "connection-low-water" => 75,
+        "console-enabled" => true,
+        "constant-headers" => undefined,
+        "http-authentication-factory" => "management-http-authentication",
+        "http-upgrade" => {
+            "enabled" => true,
+            "sasl-authentication-factory" => "management-sasl-authentication"
+        },
+        "http-upgrade-enabled" => true,
+        "no-request-timeout" => 60000,
+        "sasl-protocol" => "remote",
+        "secure-socket-binding" => undefined,
+        "server-name" => undefined,
+        "socket-binding" => "management-http",
+        "ssl-context" => undefined
+    }
+}
+----
+
+When persisted to the configuration file the resulting XML will look like:
+
+[source,xml]
+----
+<management-interfaces>
+    <http-interface http-authentication-factory="management-http-authentication"
+        backlog="51" no-request-timeout="60001"
+        connection-high-water="101" connection-low-water="76">
+
+        <http-upgrade enabled="true" sasl-authentication-factory="management-sasl-authentication"/>
+        <socket-binding http="management-http"/>
+    </http-interface>
+</management-interfaces>
+----
+
+=== Nice-to-Have Requirements
+
+None
+
+=== Non-Requirements
+
+The purpose of this feature request is to expose management model attributes to configure existing behaviour within the
+application server, this feature request is not intended to revisit the existing behaviour which is already available at
+all stability levels.
+
+=== Future Work
+
+The only possible future work could be to promote one or all of these new attributes to a higher stability level.
+
+A change such as this should likely be constrained to:
+
+ * Removing the stability level qualifier from the attribute definitions allowing them to be used at the default stability level.
+ * Moving the XML schema definitions and parsing to an appropriate schema at the default stability level.
+
+== Backwards Compatibility
+
+In considering backwards compatibility if an attribute is defined on the HTTP management interface AND the
+corresponding system property is set then the system property should take precedence.
```

Comment:
If this happens we should log a WARN that the system property will be removed. People should move promptly off of system property based configuration once proper attributes are available.

And then remove it in, say, WF 36, assuming this lands in WF 34.

Probably using the same logger category that we use when warning about use of deprecated management API would be good. It's an analogous situation.

Reply:
I have updated to add the logging:

https://github.com/wildfly/wildfly-proposals/pull/595/files#diff-c4fba9f719d2a6d63183999eb1869081774e7d44fb5d70004133b12de65c8dd6R176

The other deprecation messages all appear to use INFO so I have done the same here.

Reply:
Thanks. LGTM.